# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: c6db418af190
# Full Hash: c6db418af19097f562d94b87bbf25362556b2fbc
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-10-21 08:39:18
# Regressor Bug: 1787284
# File Overlap Count: 1
# Description:
#   Bug 1787284: [Part 2] Cache name and create radio name pivot rule r=eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159118
# ==============================================================================

diff -r fda6706bb580 -r c6db418af190 accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Thu Oct 20 21:40:05 2022 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Thu Oct 20 21:40:06 2022 +0000
@@ -3595,6 +3595,21 @@
   }
 
   if (aCacheDomain & CacheDomain::Relations && mContent) {
+    if (IsHTMLRadioButton()) {
+      // HTML radio buttons with the same name should be grouped
+      // and returned together when their MEMBER_OF relation is
+      // requested. We cache the name attribute, if it exists, here.
+      nsString name;
+      mContent->AsElement()->GetAttr(kNameSpaceID_None, nsGkAtoms::name, name);
+      if (!name.IsEmpty()) {
+        fields->SetAttribute(nsGkAtoms::radioLabel, std::move(name));
+      } else if (aUpdateType != CacheUpdateType::Initial) {
+        // It's possible we used to have a name and it's since been
+        // removed. Send a delete entry.
+        fields->SetAttribute(nsGkAtoms::radioLabel, DeleteEntry());
+      }
+    }
+
     for (auto const& data : kRelationTypeAtoms) {
       nsTArray<uint64_t> ids;
       nsStaticAtom* const relAtom = data.mAtom;